android/2023-emmsale/app/src/main/java/com/emmsale/presentation/ui/main/event/conferenceFilter/ConferenceFilterActivity.kt

```diff
@@ -108,30 +107,20 @@ class ConferenceFilterActivity : AppCompatActivity() {
 
     private fun setupEventFilters() {
         viewModel.eventFilters.observe(this) { eventFilters ->
-            when (eventFilters) {
-                is ConferenceFiltersUiState.Success -> {
-                    updateFilterViews(eventFilters)
-                    binding.progressbarLoading.visibility = View.GONE
-                }
-
-                is ConferenceFiltersUiState.Error -> {
-                    showToast(getString(R.string.all_data_loading_failed_message))
-                    binding.progressbarLoading.visibility = View.GONE
-                }
-
-                is ConferenceFiltersUiState.Loading ->
-                    binding.progressbarLoading.visibility = View.VISIBLE
+            when {
+                eventFilters.isError -> showToast(getString(R.string.all_data_loading_failed_message))
+                !eventFilters.isLoading && !eventFilters.isError -> updateFilterViews(eventFilters)
             }
```

Comment:
when 문은 위의 조건부터 차례차례 검증하기 때문에, 아래 조건들은 위의 조건을 만족하지 않아야만 실행될 수 있어요.
따라서 
```suggestion
            when {
                eventFilters.isError -> showToast(getString(R.string.all_data_loading_failed_message))
                !eventFilters.isLoading -> updateFilterViews(eventFilters)
            }
```
이렇게 써도 될 것 같아요!

Reply:
사실 처음에는 가독 측면에서 두 조건을 추가했었습니다.
하지만 스캇 리뷰를 반영하면서, 이미 아래와 같이 변경되었습니다! 🤣